elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/AsyncIT.java

```diff
@@ -153,59 +313,180 @@ public void jsonApiRequestTests() throws InterruptedException {
                                         )
                                 )
-                        )
-                ).toResponse();
+                        ).toJSON())
+                .when()
+                .post("/asyncQuery")
+                .then()
+                .statusCode(org.apache.http.HttpStatus.SC_CREATED)
+                .body("data.id", equalTo("edc4a871-dff2-4054-804e-d80075cf828e"))
+                .body("data.type", equalTo("asyncQuery"))
+                .body("data.attributes.status", equalTo("PROCESSING"))
+                .body("data.attributes.result.contentLength", nullValue())
+                .body("data.attributes.result.responseBody", nullValue())
+                .body("data.attributes.result.httpStatus", nullValue())
+                .body("data.attributes.result.resultType", nullValue());
+
+        int i = 0;
+        while (i < 1000) {
+            Thread.sleep(10);
+            Response response = given()
+                    .accept("application/vnd.api+json")
+                    .get("/asyncQuery/edc4a871-dff2-4054-804e-d80075cf828e");
+
+            // If Async Query is created and completed then validate results
+            if (response.jsonPath().getString("data.attributes.status").equals("COMPLETE")) {
+
+                // Validate AsyncQuery Response
+                response
+                        .then()
+                        .statusCode(HttpStatus.SC_OK)
+                        .body("data.id", equalTo("edc4a871-dff2-4054-804e-d80075cf828e"))
+                        .body("data.type", equalTo("asyncQuery"))
+                        .body("data.attributes.queryType", equalTo("GRAPHQL_V1_0"))
+                        .body("data.attributes.status", equalTo("COMPLETE"));
+
+                // Validate AsyncQueryResult Response
+                given()
+                        .accept("application/vnd.api+json")
+                        .get("/asyncQuery/edc4a871-dff2-4054-804e-d80075cf828e")
+                        .then()
+                        .statusCode(HttpStatus.SC_OK)
+                        .body("data.id", equalTo("edc4a871-dff2-4054-804e-d80075cf828e"))
+                        .body("data.type", equalTo("asyncQuery"))
+                        .body("data.attributes.result.contentLength", notNullValue())
+                        .body("data.attributes.result.responseBody", equalTo("{\"data\":{\"book\":{\"edges\":"
+                                + "[{\"node\":{\"id\":\"1\",\"title\":\"Ender's Game\"}},"
+                                + "{\"node\":{\"id\":\"2\",\"title\":\"Song of Ice and Fire\"}},"
+                                + "{\"node\":{\"id\":\"3\",\"title\":\"For Whom the Bell Tolls\"}}]}}}"))
+                        .body("data.attributes.result.httpStatus", equalTo(200))
+                        .body("data.attributes.result.resultType", equalTo(ResultType.EMBEDDED.toString())).toString();
+
+                // Validate GraphQL Response
```

Comment:
I don't understand the validate GraphQL response here.  Why are we doing this for each test?

Reply:
This part is querying for the asyncquery in graphql and checking for the embedded result. There are separate json api and graphql endpoint tests in AsyncIT